src/main/java/com/xiaomi/infra/pegasus/operator/client_operator.java

```diff
@@ -10,18 +10,19 @@
 import com.xiaomi.infra.pegasus.thrift.TException;
 
 public abstract class client_operator {
-  public client_operator(gpid gpid, String tableName) {
+  public client_operator(gpid gpid, String tableName, boolean isWrite) {
```

Comment:
```java
class ClientOperatorTypeHelper {
  static bool isWrite(final rrdb_check_and_set_operator op) {
    return true;
  }
  static bool isWrite(final rrdb_get_operator op) {
    return false;
  }
  static bool isWrite(final rrdb_put_operator op) {
    return true;
  }
}
```
I think this better.

Reply:
Yes this way need few modify, but if we want to add a new operator, we also should add a function in ClientOperatorTypeHelper. If somebody forget to add this function, that will introduce a disater.  conversely, if we use write flag in operator, one must set it, or it will produce compile error.
Moreover, I think write or read is a feature of operator, using another class to manage it is a little tricky. Besides, add a flag is once and for all.